pkg/controllerutil/predicate.go

```diff
@@ -56,3 +131,27 @@ func namespacePredicateFilter(object client.Object) bool {
 	}
 	return managedNamespaces.Has(object.GetNamespace())
 }
+
+func newAPIVersionPredicateFilter(objs []client.Object) func(client.Object) bool {
+	return func(obj client.Object) bool {
+		annotations := obj.GetAnnotations()
+		if annotations != nil {
+			apiVersion, ok := annotations[constant.CRDAPIVersionAnnotationKey]
+			if ok {
+				return IsSupportedCRDAPIVersion(apiVersion)
+			}
+		}
+		switch reflect.TypeOf(obj) {
+		case reflect.TypeOf(&appsv1.Cluster{}):
+			return true
+		case reflect.TypeOf(&appsv1.ClusterDefinition{}),
+			reflect.TypeOf(&appsv1.ComponentDefinition{}),
+			reflect.TypeOf(&appsv1.ComponentVersion{}),
+			reflect.TypeOf(&appsv1.Component{}),
+			reflect.TypeOf(&workloadsv1.InstanceSet{}):
```

Comment:
how to reconcile the cd/cmpd,  seems` constant.CRDAPIVersionAnnotationKey` not patched on these objects

Reply:
Annotations will be added to cd/cmpd/cmpv within the addon.

Reply:
and is only needed when coexisting with version 0.9